File: Python/Warmup-2/count_xx.py

# this function return the number of "xx" in the given string
# overlapping is allowed
def count_xx(str):
	count = 0

	for x in range(len(str)-1):
		if str[x] == 'x' and str[x+1] == 'x':
			count += 1

	return count

print(count_xx("abcxx"))
print(count_xx("xxx"))
print(count_xx("xxxx"))

File: JavaScript/Warmup-2/countXX.js

// count the number of "xx" in the given string
// we'll say that overlapping is allowed, so "xxx" contains 2 "xx"
const countXX = (str) => {

    let count = 0;

    for (let i = 0; i < str.length - 1; i++) {
        if (str[i].toLowerCase() === 'x' && str[i+1].toLowerCase() === 'x') {
            count++;
        }
    }

    return count;

};

console.log(countXX('abcxx'));
console.log(countXX('xxx'));
console.log(countXX('xxxx'));
